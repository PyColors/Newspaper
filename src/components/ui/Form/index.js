import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Form = styled.form`
  padding: 15px;
`;

class FormComponent extends Component {
  render() {
    const { id, onSubmit, action, method } = this.props;

    return <Form id={id} onSubmit={onSubmit} action={action} method={method} />;
  }
}

Form.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  id: PropTypes.string,
  action: PropTypes.string,
  method: PropTypes.string
};

Form.defaultProps = {
  id: '',
  action: '',
  method: ''
};
export default FormComponent;
