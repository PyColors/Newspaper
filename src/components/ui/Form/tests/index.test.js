import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Form from '../index';

describe('<Form />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer
      .create(
        <Form
          id={'id'}
          onSubmit={'onSubmit'}
          action={'action'}
          method={'method'}
        />
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render a prop', () => {
    const id = 'testH1Id';
    const renderedComponent = shallow(<Form id={id} />);
    expect(renderedComponent.prop('id')).toEqual(id);
  });
});
