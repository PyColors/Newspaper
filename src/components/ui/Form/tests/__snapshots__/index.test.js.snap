// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should match the snapshot 1`] = `
<form
  action="action"
  className="sc-bdVaJa HuctC"
  id="id"
  method="method"
  onSubmit="onSubmit"
/>
`;
